packages/graph-engine/src/nodes/color/lib/sortColors.ts

```diff
@@ -14,25 +13,22 @@ export const compareFunctions = {
 	Distance: (foreground, background) => foreground.deltaE(background, '2000')
 };
 
-export const sortTokens = (colors, compareColor, type, algorithm) =>
-	orderBy(
-		colors.map((color, index) => {
+export const sortTokens = (colors, compareColor, type, algorithm) => {
+	const background = toColor(compareColor);
+	const compareFunction = compareFunctions[type];
+
+	return colors
+		.reduce((acc, color, index) => {
 			const foreground = toColor(color);
-			const background = toColor(compareColor);
-			const compareValue = compareFunctions[type](
-				foreground,
-				background,
-				algorithm
-			);
+			const compareValue = compareFunction(foreground, background, algorithm);
 
-			return {
+			acc.push({
 				color,
 				compareValue,
 				index
-			};
-		}),
-		['compareValue']
-	).map(item => ({
-		color: item.color,
-		index: item.index
-	}));
+			});
+			return acc;
+		}, [])
+		.sort((a, b) => a.compareValue - b.compareValue)
+		.map(({ color, index }) => ({ color, index }));
```

Comment:
You're still iterating over the array 3 times, just do a single reduce here and return 2 arrays, one with the color and the indices so you don't need to map over it 2 more times to extract them